File: cpu_usage_indicator/matlab/cpu_usage_indicator.m
ledr = "/sys/class/leds/sunxi_led0r/brightness";
ledg = "/sys/class/leds/sunxi_led0g/brightness";

obj_clean_up = onCleanup(@()clean_up(ledr, ledg));

led_set_brightness(ledr, 0);
led_set_brightness(ledg, 0);

[stat_used, stat_total] = proc_stat_get_used_and_total();

while 1
  pause(0.8);

  led_set_brightness(ledr, 0);
  led_set_brightness(ledg, 0);
  
  pause(0.2);

  [latest_used, latest_total] = proc_stat_get_used_and_total();

  cpu_usage = (latest_used - stat_used) * 100 / (latest_total - stat_total);
  cpu_usage = round(cpu_usage);
  
  fprintf(stdout, "\rcpu usage = %d %%  ", cpu_usage);
  fflush(stdout);
  
  stat_used = latest_used;
  stat_total = latest_total;
  
  if (cpu_usage < 30)
    led_set_brightness(ledg, 30);
  else 
    if (cpu_usage < 60)
      led_set_brightness(ledg, 5);
    else 
      if cpu_usage < 90
        led_set_brightness(ledr, 15);
      else
        led_set_brightness(ledr, 150);
      end
    end
  end
  
end

File: cpu_usage_indicator/go/cpu_usage_indicator.go
package main

import "fmt"
import "time"
import "os"
import "os/signal"


type app_t struct {
	running chan os.Signal
	usage uint8
	ledr led_t
	ledg led_t
	stat proc_stat_t
}


func main() {
	var app app_t

	app.running = make(chan os.Signal, 1)
	signal.Notify(app.running, os.Interrupt, os.Kill)

	proc_stat_init(&app.stat)
	led_init(&app.ledr, "/sys/class/leds/sunxi_led0r/brightness")
	led_init(&app.ledg, "/sys/class/leds/sunxi_led0g/brightness")

main_loop:
	for true {
		time.Sleep(time.Millisecond * 800)

		led_set_brightness(&app.ledr, 0)
		led_set_brightness(&app.ledg, 0)

		time.Sleep(time.Millisecond * 200)

		app.usage = proc_stat_get_cpu_usage(&app.stat)
		fmt.Printf("\rcpu usage = %d %%  ", app.usage)

		if app.usage < 30 {
			led_set_brightness(&app.ledg, 30)
		} else if app.usage < 60 {
			led_set_brightness(&app.ledg, 5)
		} else if app.usage < 90 {
			led_set_brightness(&app.ledr, 15)
		} else {
			led_set_brightness(&app.ledr, 150)
		}

		select {
			case <- app.running:
				break main_loop
			default:
		}
	}

	fmt.Printf("\n")
	led_fini(&app.ledr)
	led_fini(&app.ledg)
}